File: typescript/resistor-color/resistor-color.ts

const colorMap = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
}

export const COLORS: string[] =  [...Object.keys(colorMap)];


export function colorCode(color: keyof typeof colorMap): number {
  return colorMap[color];
}

File: c/resistor-color/resistor_color.c
#include "resistor_color.h"

int color_code(resistor_band_t resistor_band) {
   return resistor_band;
}

resistor_band_t * colors(void) {
   static resistor_band_t colors[10];
   for (resistor_band_t color = BLACK; color <= WHITE; color++) {
      colors[color] = color;
   }
   return colors;
}
